project/src/test/java/com/google/sps/BlobFormHandlerTest.java

```diff
@@ -0,0 +1,88 @@
+package com.google.sps;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.sps.data.User;
+import com.google.sps.servlets.BlobFormHandlerServlet;
+import com.google.sps.servlets.BlobUrlServlet;
+import com.google.sps.utilities.CommonUtils;
+import com.google.sps.utilities.SpannerClient;
+import com.google.sps.utilities.SpannerTasks;
+import com.google.sps.utilities.SpannerTestTasks;
+import com.google.sps.utilities.TestUtils;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockServletContext;
+
+/** Unit tests for DatabaseWrapper functionality related to Event class. */
+@RunWith(JUnit4.class)
+public class BlobFormHandlerTest {
```

Comment:
Also test the failures of the Servlet?

Reply:
Done, tests invalid users now to make sure the servlet sends the right error

Reply:
Could you also test the case when MalformedURLException is thrown?